### PR TITLE
Flow types

### PR DESCRIPTION
**Pull request by kramarmm**
Added checking types with flow

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,14 +36,17 @@
     "babel-eslint": "6.0.4",
     "babel-loader": "6.2.4",
     "babel-preset-es2015": "6.6.0",
+    "babel-preset-flow": "^6.23.0",
```
Comment (max-mykhailenko) [inline package.json]: Version should be exact (without ^)

File: package.json [hunk 2]
```diff
@@ -35,14 +36,17 @@
...
     "babel-preset-react": "6.3.13",
     "babel-preset-stage-1": "6.5.0",
     "conventional-changelog": "1.1.0",
     "eslint": "2.9.0",
     "eslint-config-airbnb": "^9.0.0",
+    "eslint-plugin-flowtype": "^2.34.0",
```
Comment (max-mykhailenko) [inline package.json]: Same here

File: src/format.js [hunk 3]
```diff
@@ -84,160 +85,161 @@ export default function date(value, format, utc = false) {
...
     },
 
     // Year
-    L() {
+    L(): number {
       // Is leap year?; 0 or 1
-      const j = f.Y();
-      return j % 4 === 0 & j % 100 !== 0 | j % 400 === 0;
+      const j: number = f.Y();
+      return (j % 4 === 0 && j % 100 !== 0 || j % 400 === 0) ? 1 : 0;
```
Comment (max-mykhailenko) [inline src/format.js]: Why are you add additional `&` and `|`?

Comment (kramarmm) [inline src/format.js]: I had this error:
image
and i noticed that below  is using `? 1 : 0`,
so, i decided that it shoud be the same, if I replace bitwise `& |` to ternary statement that returns the same.
Or it doesnt returns the same?

Comment (max-mykhailenko) [inline src/format.js]: I think this change require additional tests.

Comment (kramarmm) [inline src/format.js]: additional tests:
https://github.com/kramarmm/react-pure-time/commit/0c367766c433923d99d4bb386492482a435dd871#diff-e5060184b5c2c8908fcfcfeb5f36bfdd

File: src/format.js [hunk 4]
```diff
@@ -1,24 +1,25 @@
+// @flow
 // based on phpjs date format library
 
-export default function date(value, format, utc = false) {
-  const jsdate = new Date(value);
+export default function date(value: Date, format: string, utc: boolean = false): string {
```
Comment (max-mykhailenko) [inline src/format.js]: Здесь не только Date раз мы ниже делаем new Date(value)

**Comment by max-mykhailenko** [comment]
@kramarmm PR is waiting for you )

**Pull request closed without merging**